src/main/groovy/liquibase/harness/data/ChangeDataTest.groovy

```diff
@@ -0,0 +1,97 @@
+package liquibase.harness.data
+
+import liquibase.Liquibase
+import liquibase.database.jvm.JdbcConnection
+import liquibase.harness.config.TestConfig
+import org.json.JSONObject
+import org.junit.Assert
+import org.junit.Assume
+import spock.lang.Specification
+import spock.lang.Unroll
+
+import static liquibase.harness.util.JSONUtils.*
+import static liquibase.harness.util.FileUtils.*
+import static liquibase.harness.util.TestUtils.*
+import static ChangeDataTestHelper.*
+
+class ChangeDataTest extends Specification {
+
+    @Unroll
+    def "apply #testInput.changeData against #testInput.databaseName, #testInput.version; verify generated query, checking query and obtained result set"() {
+
+        Liquibase liquibase = createLiquibase(testInput.pathToChangeLogFile, testInput.database)
+        given: "create Liquibase connection, read expected sql and obtain result set"
+        String expectedSql = cleanSql(getExpectedSqlFileContent(testInput.changeData,
+                testInput.databaseName, testInput.version, "liquibase/harness/data/expectedSql"))
+        String checkingSql = cleanSql(getExpectedSqlFileContent(testInput.changeData,
+                testInput.databaseName, testInput.version, "liquibase/harness/data/checkingSql"))
+        String expectedResultSet = getExpectedJSONFileContent(testInput.changeData, testInput.databaseName,
+                testInput.version, "liquibase/harness/data/expectedResultSet")
+
+        and: "skip testcase if it's invalid for this combination of db type and/or version"
+        Assume.assumeTrue(expectedSql, expectedSql == null || !expectedSql.toLowerCase().contains("invalid test"))
+
+        and: "fail test if expectedResultSet is not provided"
+        assert expectedResultSet != null: "No expectedResultSet for ${testInput.changeData} against " +
+                "${testInput.database.shortName} " +
+                "${testInput.database.databaseMajorVersion}.${testInput.database.databaseMinorVersion}"
+
+        and: "fail test if checkingSql is not provided"
+        assert checkingSql != null: "No checkingSql for ${testInput.changeData} against " +
+                "${testInput.database.shortName} " +
+                "${testInput.database.databaseMajorVersion}.${testInput.database.databaseMinorVersion}"
+
+        when: "get sql that is generated for changeset"
+        def generatedSql = cleanSql(toSqlFromLiquibaseChangeSets(liquibase))
+
+        then: "verify expected sql matches generated sql"
+        if (expectedSql != null && !testInput.pathToChangeLogFile.endsWith(".sql")) {
+            assert generatedSql == expectedSql: "Expected SQL does not match actual sql. " +
```

Comment:
nit
mixing uppercase and lowercase "SQL"

Reply:
++